libs/datamodel/core/tests/reformat/reformat_implicit_relations.rs

```diff
@@ -362,42 +972,34 @@ model Post {
 }
 "#;
 
-    assert_reformat(input, expected);
+    let result = datamodel::ast::reformat::Reformatter::new(&input).reformat_to_string();
+
+    assert_eq!(result, expected);
 }
 
+//todo formatter failure??
 #[test]
-#[ignore]
-fn must_add_relation_attribute_to_an_existing_field() {
-    let input = r#"
+fn must_handle_conflicts_with_existing_fields_if_types_are_incompatible_and_name_generation_breaks_down() {
+    //just dont add anything
+    let input = indoc! {r#"
     model Blog {
-      id    Int     @id
+      id    String @id
       posts Post[]
     }
-
+    
     model Post {
-      id     Int   @id
-      Blog   Blog? @relation(fields: [blogId])
-      blogId Int?
+      id                String @id
+      blogId            Int? // this is not compatible with Blog.id
+      blogId_BlogToPost Int? // clashes with the auto generated name
     }
-    "#;
+    "#};
 
-    let expected = r#"model Blog {
-  id    Int    @id
-  posts Post[]
-}
+    let result = datamodel::ast::reformat::Reformatter::new(&input).reformat_to_string();
 
-model Post {
-  id     Int   @id
-  Blog   Blog? @relation(fields: [blogId], references: [id])
-  blogId Int?
-}
-"#;
-    assert_reformat(input, expected);
+    assert_eq!(input, result);
 }
 
 fn assert_reformat(schema: &str, expected_result: &str) {
-    println!("schema: {:?}", schema);
     let result = datamodel::ast::reformat::Reformatter::new(&schema).reformat_to_string();
-    println!("result: {}", result);
     assert_eq!(result, expected_result);
```

Comment:
zoom comment: maybe we could validate the final result in this method? (or another?)